pkg/core/native/policy.go

```diff
@@ -0,0 +1,395 @@
+	p.AddMethod(md, desc, false)
+
+	desc = newDescriptor("setMaxTransactionsPerBlock", smartcontract.BoolType,
+		manifest.NewParameter("value", smartcontract.IntegerType))
+	md = newMethodAndPrice(p.setMaxTransactionsPerBlock, 3000000, smartcontract.NoneFlag)
+	p.AddMethod(md, desc, false)
+
+	desc = newDescriptor("setFeePerByte", smartcontract.BoolType,
+		manifest.NewParameter("value", smartcontract.IntegerType))
+	md = newMethodAndPrice(p.setFeePerByte, 3000000, smartcontract.NoneFlag)
+	p.AddMethod(md, desc, false)
+
+	desc = newDescriptor("blockAccount", smartcontract.BoolType,
+		manifest.NewParameter("account", smartcontract.Hash160Type))
+	md = newMethodAndPrice(p.blockAccount, 3000000, smartcontract.NoneFlag)
+	p.AddMethod(md, desc, false)
+
+	desc = newDescriptor("unblockAccount", smartcontract.BoolType,
+		manifest.NewParameter("account", smartcontract.Hash160Type))
+	md = newMethodAndPrice(p.unblockAccount, 3000000, smartcontract.NoneFlag)
+	p.AddMethod(md, desc, false)
+	return p
+}
+
+// Metadata implements Contract interface.
+func (p *Policy) Metadata() *interop.ContractMD {
+	return &p.ContractMD
+}
+
+// Initialize initializes Policy native contract and implements Contract interface.
+func (p *Policy) Initialize(ic *interop.Context) error {
+	if ic.Trigger != trigger.Application {
+		return errors.Errorf("invalid trigger type: %v", ic.Trigger.String())
+	}
+
+	si := &state.StorageItem{
+		Value: make([]byte, 4),
+	}
+	binary.LittleEndian.PutUint32(si.Value, defaultMaxBlockSize)
+	err := ic.DAO.PutStorageItem(p.Hash, maxBlockSizeKey, si)
+	if err != nil {
+		return err
+	}
+
+	si = &state.StorageItem{
+		Value: make([]byte, 4),
+	}
+	binary.LittleEndian.PutUint32(si.Value, defaultMaxTransactionsPerBlock)
+	err = ic.DAO.PutStorageItem(p.Hash, maxTransactionsPerBlockKey, si)
+	if err != nil {
+		return err
+	}
+
+	si = &state.StorageItem{
+		Value: make([]byte, 8),
+	}
+	binary.LittleEndian.PutUint64(si.Value, defaultFeePerByte)
+	err = ic.DAO.PutStorageItem(p.Hash, feePerByteKey, si)
+	if err != nil {
+		return err
+	}
+
+	ba := new(BlockedAccounts)
+	si = &state.StorageItem{
+		Value: ba.Bytes(),
+	}
+	err = ic.DAO.PutStorageItem(p.Hash, blockedAccountsKey, si)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+// OnPersist implements Contract interface.
+func (p *Policy) OnPersist(ic *interop.Context) error {
+	return nil
+}
+
+// getMaxTransactionsPerBlock is Policy contract method and returns the upper
+// limit of transactions per block.
+func (p *Policy) getMaxTransactionsPerBlock(ic *interop.Context, _ []stackitem.Item) stackitem.Item {
+	return stackitem.NewBigInteger(big.NewInt(int64(p.GetMaxTransactionsPerBlockInternal(ic))))
+}
+
+// GetMaxTransactionsPerBlockInternal returns the upper limit of transactions per
+// block.
+func (p *Policy) GetMaxTransactionsPerBlockInternal(ic *interop.Context) uint32 {
+	return p.getUint32WithKey(ic, maxTransactionsPerBlockKey)
```

Comment:
And I also think that all of these values need to be cached in the global contract's state. They're small and they're rarely changed and every transaction check or block generation process needs them, so going into the DB to get uint32 makes no sense. You need to be careful with sets (disable cache if there was a set and enable it on persist (getting the actual value from the DB)) and concurrent access though.

Reply:
The other option of course is caching it at Blockchain level, but you'll have to update it on every `storeBlock` then.

Reply:
I added cache for these values right into native Policy contract. Please, take a look at 4f30f09d52e123f102819495b4598b4cb46a5902, is it possible solution? I'm not sure, that's why it's a separate commit.